src/shared/networkrequest.cpp

```diff
@@ -0,0 +1,487 @@
+
+    if (brokenUrl.host().isEmpty() && !m_redirectedUrl.isEmpty()) {
+      QUrl url = m_redirectedUrl.resolved(brokenUrl);
+
+#  ifdef MZ_DEBUG
+      // See https://bugreports.qt.io/browse/QTBUG-100651
+      logger.debug()
+          << "QT6 redirect bug! The current URL is broken because it's not "
+             "resolved using the latest HTTP redirection as base-URL";
+      logger.debug() << "Broken URL:" << brokenUrl.toString();
+      logger.debug() << "Latest redirected URL:" << m_redirectedUrl.toString();
+      logger.debug() << "Final URL:" << url.toString();
+#  endif
+
+      m_request = QNetworkRequest(url);
+      m_request.setHeader(QNetworkRequest::ContentTypeHeader,
+                          "application/json");
+      m_request.setAttribute(QNetworkRequest::RedirectPolicyAttribute,
+                             QNetworkRequest::NoLessSafeRedirectPolicy);
+
+      m_reply = nullptr;
+      m_timer.stop();
+      getRequest();
+      return;
+    }
+  }
+#endif
+
+  int status = statusCode();
+
+  QString expect =
+      m_expectedStatusCode ? QString::number(m_expectedStatusCode) : "any";
+  logger.debug() << "Network reply received - status:" << status
+                 << "- expected:" << expect;
+
+  QByteArray data = m_reply->readAll();
+  processData(m_reply->error(), m_reply->errorString(), status, data);
+}
+
+void NetworkRequest::processData(QNetworkReply::NetworkError error,
+                                 const QString& errorString, int status,
+                                 const QByteArray& data) {
+  m_completed = true;
+  m_timer.stop();
+
+#ifdef MZ_WASM
+  m_finalStatusCode = status;
+#endif
+
+  if (error != QNetworkReply::NoError) {
+    QUrl::FormattingOptions options = QUrl::RemoveQuery | QUrl::RemoveUserInfo;
+    logger.error() << "Network error:" << errorString
+                   << "status code:" << status
+                   << "- body:" << logger.sensitive(data);
+    logger.error() << "Failed to access:" << m_request.url().toString(options);
+    emit requestFailed(error, data);
+    return;
+  }
+
+  // This is an extra check for succeeded requests (status code 200 vs 201, for
+  // instance). The real network status check is done in the previous if-stmt.
+  if (m_expectedStatusCode && status != m_expectedStatusCode) {
+    logger.error() << "Status code unexpected - status code:" << status
+                   << "- expected:" << m_expectedStatusCode;
+    emit requestFailed(QNetworkReply::ConnectionRefusedError, data);
+    return;
+  }
+
+  emit requestCompleted(data);
+}
+
+bool NetworkRequest::isRedirect() const {
+  int status = statusCode();
+  return status >= 300 && status < 400;
+}
+
+void NetworkRequest::handleHeaderReceived() {
+  // Suppress this signal if a redirect is about to happen.
+  int policy =
+      m_request.attribute(QNetworkRequest::RedirectPolicyAttribute).toInt();
+  if (isRedirect() && (policy != QNetworkRequest::ManualRedirectPolicy)) {
```

Comment:
Just a question: Is this future proofing against wanting to use ManualRedirectPolicy at a later time?

Reply:
This is a workaround for a horrible bug in Qt6, fixed in 6.2: https://bugreports.qt.io/browse/QTBUG-100651
This code should be removed as soon as we force qt6.2 as min version